File: SortingInCSharp/SortingInCSharp/SelectionSort.cs
using System;
namespace SortingInCSharp
{
    public static class SelectionSort
    {

        public static int[] SortAscending(int[] array)
        {
            int minIndex;
            int len = array.Length;

            for(int i = 0; i< len; i++)
            {
                minIndex = i;
                for(int j = i+1; j< len; j++)
                {
                    if (array[j] < array[minIndex])
                    {
                        minIndex = j;
                    }
                }
                Swap(ref array, i, minIndex);
            }

            return array;
        }

        public static int[] SortDescending(int[] array)
        {
            int maxIndex = -1;
            int len = array.Length;

            for (int i = 0; i < len; i++)
            {
                maxIndex = i;
                for (int j = i + 1; j < len; j++)
                {
                    if (array[j] > array[maxIndex])
                    {
                        maxIndex = j;
                    }
                }
                Swap(ref array, i, maxIndex);
            }

            return array;
        }

        public static void Swap(ref int[] array, int index1, int index2)
    {
        int temp = array[index1];
        array[index1] = array[index2];
        array[index2] = temp;

    }
}
}

File: SortingInPython/SelectionSort.py

def sortAscending(array):
    l = len(array)
    minIndex = -1
    for i in range(l):
        minIndex = i
        for j in range(i+1, l):
            if array[j] < array[minIndex]:
                minIndex = j
        
        swap(array, i, minIndex)



def sortDescending(array):
    l = len(array)
    minIndex = -1
    for i in range(l):
        minIndex = i
        for j in range(i+1, l):
            if array[j] > array[minIndex]:
                minIndex = j
        
        swap(array, i, minIndex)




def swap(array, index1, index2):
    temp = array[index1]
    array[index1] = array[index2]
    array[index2] = temp


def display(array):
    l = len(array)
    for x in range(l):
        print(array[x])




array = [-2, 45, -12, 11, -9 ]
sortDescending(array)

display(array)